kc export fails when using User Federation (LDAP) with SSL/TLS

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

import-export

### Describe the bug

Since the issue (https://github.com/keycloak/keycloak/issues/22617) is included in this morning's version 24.0.4, I wanted to test this change right away, but this resulted in a new error :/, in export mode there are probably problems with the Truststore

### Version

24.0.4

### Regression

- [ ] The issue is a regression

### Expected behavior

kc export with users should work when LDAP federation with SSL/TLS and file-based vault is used.

### Actual behavior

```
2024-05-08 10:57:42.213  Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target 
 2024-05-08 10:57:42.213 at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:388) 
 2024-05-08 10:57:42.213 at java.base/sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:271) 
 2024-05-08 10:57:42.213 at java.base/sun.security.validator.Validator.validate(Validator.java:256) 
 2024-05-08 10:57:42.213 at java.base/sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:230) 
 2024-05-08 10:57:42.213 at java.base/sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:132) 
 2024-05-08 10:57:42.213 at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(CertificateMessage.java:631) 
 2024-05-08 10:57:42.213 ... 108 more 
```

Full stack trace:

```
 10:57:32.640  d:\keycloak>.\keycloak-24.0.4\bin\kc.bat export --verbose --users same_file --file export/realmName-2024-05-08-10-57-32.json --realm realmName  
 10:57:38.492  10:57:35,672 INFO  [org.keycloak.quarkus.runtime.hostname.DefaultHostnameProvider] (main) Hostname settings: Base URL: <unset>, Hostname: <request>, Strict HTTPS: false, Path: <request>, Strict BackChannel: false, Admin URL: <unset>, Admin: <request>, Port: -1, Proxied: false 
 10:57:38.493  10:57:35,964 WARN  [org.infinispan.CONFIG] (keycloak-cache-init) ISPN000569: Unable to persist Infinispan internal caches as no global state enabled 
 10:57:38.494  10:57:36,079 INFO  [org.infinispan.CONTAINER] (keycloak-cache-init) ISPN000556: Starting user marshaller 'org.infinispan.jboss.marshalling.core.JBossUserMarshaller' 
 10:57:38.558  10:57:38,558 WARN  [io.quarkus.agroal.runtime.DataSources] (JPA Startup Thread) Datasource <default> enables XA but transaction recovery is not enabled. Please enable transaction recovery by setting quarkus.transaction-manager.enable-recovery=true, otherwise data may be lost if the application is terminated abruptly 
 10:57:39.566  10:57:39,566 INFO  [org.keycloak.connections.infinispan.DefaultInfinispanConnectionProviderFactory] (main) Node name: node_108892, Site name: null 
 10:57:39.573  10:57:39,573 INFO  [org.keycloak.broker.provider.AbstractIdentityProviderMapper] (main) Registering class org.keycloak.broker.provider.mappersync.ConfigSyncEventListener 
 10:57:41.096  10:57:41,096 INFO  [org.keycloak.services] (main) KC-SERVICES0034: Export of realm 'realmName' requested. 
 10:57:41.096  10:57:41,096 INFO  [org.keycloak.exportimport.singlefile.SingleFileExportProvider] (main) Exporting realm 'realmName' into file d:\keycloak\export\realmName-2024-05-08-10-57-32.json 
 10:57:41.967  10:57:41,967 INFO  [org.keycloak.storage.ldap.LDAPIdentityStoreRegistry] (main) Creating new LDAP Store for the LDAP storage provider: 'ldap', LDAP Configuration: {serverPrincipal=[HTTP/ldap.company.local@COMPANY.LOCAL], fullSyncPeriod=[-1], pagination=[true], startTls=[false], connectionPooling=[true], usersDn=[DC=company,DC=local], cachePolicy=[NO_CACHE], useKerberosForPasswordAuthentication=[true], importEnabled=[true], enabled=[true], changedSyncPeriod=[-1], bindDn=[admin@COMPANY.LOCAL], usernameLDAPAttribute=[sAMAccountName], vendor=[ad], uuidLDAPAttribute=[objectGUID], allowKerberosAuthentication=[true], connectionUrl=[ldaps://ldap.company.local:636], syncRegistrations=[false], authType=[simple], debug=[true], searchScope=[2], keyTab=[d:\keycloak\my.keytab9], useTruststoreSpi=[always], usePasswordModifyExtendedOp=[false], priority=[0], kerberosRealm=[COMPANY.LOCAL], trustEmail=[false], userObjectClasses=[person, organizationalPerson, user], rdnLDAPAttribute=[cn], editMode=[READ_ONLY], validatePasswordPolicy=[false], batchSizeForSync=[1000]}, binaryAttributes: [] 
 10:57:42.209  10:57:42,209 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Failed to start server in (import_export) mode 
 10:57:42.212  10:57:42,209 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) Error details:: org.keycloak.models.ModelException: LDAP Query failed 
 10:57:42.212  10:57:42,209 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) Error details:: org.keycloak.models.ModelException: LDAP Query failed 
 10:57:42.212 at org.keycloak.storage.ldap.idm.query.internal.LDAPQuery.getResultList(LDAPQuery.java:178) 
 10:57:42.212 at org.keycloak.storage.ldap.idm.query.internal.LDAPQuery.getFirstResult(LDAPQuery.java:185) 
 10:57:42.212 at org.keycloak.storage.ldap.LDAPStorageProvider.loadLDAPUserByUuid(LDAPStorageProvider.java:1024) 
 10:57:42.212 at org.keycloak.storage.ldap.LDAPStorageProvider.loadAndValidateUser(LDAPStorageProvider.java:621) 
 10:57:42.212 at org.keycloak.storage.ldap.LDAPStorageProvider.validate(LDAPStorageProvider.java:192) 
 10:57:42.212 at org.keycloak.storage.UserStorageManager.importValidation(UserStorageManager.java:133) 
 10:57:42.212 at org.keycloak.storage.UserStorageManager.getUserById(UserStorageManager.java:372) 
 10:57:42.212 at org.keycloak.models.cache.infinispan.UserCacheSession.getUserById(UserCacheSession.java:219) 
 10:57:42.212 at org.keycloak.models.jpa.JpaUserProvider.lambda$searchForUserStream$6(JpaUserProvider.java:761) 
 10:57:42.212 at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) 
 10:57:42.212 at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179) 
 10:57:42.212 at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133) 
 10:57:42.212 at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1939) 
 10:57:42.212 at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) 
 10:57:42.212 at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) 
 10:57:42.212 at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151) 
 10:57:42.212 at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174) 
 10:57:42.212 at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) 
 10:57:42.212 at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596) 
 10:57:42.212 at org.keycloak.utils.ClosingStream.forEach(ClosingStream.java:128) 
 10:57:42.212 at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276) 
 10:57:42.212 at java.base/java.util.stream.Streams$StreamBuilderImpl.forEachRemaining(Streams.java:411) 
 10:57:42.212 at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:734) 
 10:57:42.212 at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:734) 
 10:57:42.212 at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) 
 10:57:42.212 at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) 
 10:57:42.212 at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) 
 10:57:42.212 at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) 
 10:57:42.212 at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) 
 10:57:42.212 at org.keycloak.exportimport.util.ExportUtils.exportRealm(ExportUtils.java:219) 
 10:57:42.212 at org.keycloak.exportimport.util.ExportUtils.exportRealm(ExportUtils.java:73) 
 10:57:42.212 at org.keycloak.exportimport.singlefile.SingleFileExportProvider$2.runExportImportTask(SingleFileExportProvider.java:93) 
 10:57:42.212 at org.keycloak.exportimport.util.ExportImportSessionTask.run(ExportImportSessionTask.java:35) 
 10:57:42.212 at org.keycloak.models.utils.KeycloakModelUtils.lambda$runJobInTransaction$1(KeycloakModelUtils.java:257) 
 10:57:42.212 at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransactionWithResult(KeycloakModelUtils.java:379) 
 10:57:42.212 at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransaction(KeycloakModelUtils.java:256) 
 10:57:42.212 at org.keycloak.exportimport.singlefile.SingleFileExportProvider.exportRealm(SingleFileExportProvider.java:86) 
 10:57:42.212 at org.keycloak.exportimport.singlefile.SingleFileExportProvider.exportModel(SingleFileExportProvider.java:65) 
 10:57:42.212 at org.keycloak.exportimport.ExportImportManager.runExport(ExportImportManager.java:163) 
 10:57:42.212 at org.keycloak.services.resources.KeycloakApplication.startup(KeycloakApplication.java:151) 
 10:57:42.212 at org.keycloak.quarkus.runtime.integration.jaxrs.QuarkusKeycloakApplication.onStartupEvent(QuarkusKeycloakApplication.java:54) 
 10:57:42.212 at org.keycloak.quarkus.runtime.integration.jaxrs.QuarkusKeycloakApplication_Observer_onStartupEvent_GNZ8m5QenZ9h9VNelo7awjUZFDE.notify(Unknown Source) 
 10:57:42.212 at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:346) 
 10:57:42.212 at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:328) 
 10:57:42.212 at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:82) 
 10:57:42.212 at io.quarkus.arc.runtime.ArcRecorder.fireLifecycleEvent(ArcRecorder.java:155) 
 10:57:42.212 at io.quarkus.arc.runtime.ArcRecorder.handleLifecycleEvents(ArcRecorder.java:106) 
 10:57:42.212 at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy_0(Unknown Source) 
 10:57:42.212 at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy(Unknown Source) 
 10:57:42.212 at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source) 
 10:57:42.212 at io.quarkus.runtime.Application.start(Application.java:101) 
 10:57:42.212 at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111) 
 10:57:42.212 at io.quarkus.runtime.Quarkus.run(Quarkus.java:71) 
 10:57:42.212 at org.keycloak.quarkus.runtime.KeycloakMain.start(KeycloakMain.java:108) 
 10:57:42.212 at org.keycloak.quarkus.runtime.cli.command.AbstractStartCommand.run(AbstractStartCommand.java:33) 
 10:57:42.212 at org.keycloak.quarkus.runtime.cli.command.AbstractExportImportCommand.run(AbstractExportImportCommand.java:47) 
 10:57:42.212 at picocli.CommandLine.executeUserObject(CommandLine.java:2026) 
 10:57:42.212 at picocli.CommandLine.access$1500(CommandLine.java:148) 
 10:57:42.212 at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461) 
 10:57:42.212 at picocli.CommandLine$RunLast.handle(CommandLine.java:2453) 
 10:57:42.212 at picocli.CommandLine$RunLast.handle(CommandLine.java:2415) 
 10:57:42.212 at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273) 
 10:57:42.212 at picocli.CommandLine$RunLast.execute(CommandLine.java:2417) 
 10:57:42.212 at picocli.CommandLine.execute(CommandLine.java:2170) 
 10:57:42.212 at org.keycloak.quarkus.runtime.cli.Picocli.parseAndRun(Picocli.java:125) 
 10:57:42.212 at org.keycloak.quarkus.runtime.KeycloakMain.main(KeycloakMain.java:98) 
 10:57:42.212 at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) 
 10:57:42.212 at java.base/java.lang.reflect.Method.invoke(Method.java:580) 
 10:57:42.212 at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:62) 
 10:57:42.212 at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:33) 
 10:57:42.212  Caused by: org.keycloak.models.ModelException: Querying of LDAP failed org.keycloak.storage.ldap.idm.query.internal.LDAPQuery@3aef4650 
 10:57:42.212 at org.keycloak.storage.ldap.idm.store.ldap.LDAPIdentityStore.fetchQueryResults(LDAPIdentityStore.java:295) 
 10:57:42.212 at org.keycloak.storage.ldap.idm.query.internal.LDAPQuery.getResultList(LDAPQuery.java:174) 
 10:57:42.212 ... 69 more 
 10:57:42.212  Caused by: org.keycloak.models.ModelException: Could not query server using DN [DC=company,DC=local] and filter [(objectGUID=\42\1c\f0\b3\f4\66\1d\45\b6\7a\fe\f4\55\a4\b3\bc)] 
 10:57:42.212 at org.keycloak.storage.ldap.idm.store.ldap.LDAPOperationManager.lookupById(LDAPOperationManager.java:434) 
 10:57:42.212 at org.keycloak.storage.ldap.idm.store.ldap.LDAPIdentityStore.fetchQueryResults(LDAPIdentityStore.java:261) 
 10:57:42.212 ... 70 more 
 10:57:42.212  Caused by: javax.naming.CommunicationException: ldap.company.local:636 [Root exception is javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target] 
 10:57:42.212 at java.naming/com.sun.jndi.ldap.Connection.<init>(Connection.java:253) 
 10:57:42.212 at java.naming/com.sun.jndi.ldap.LdapClient.<init>(LdapClient.java:141) 
 10:57:42.212 at java.naming/com.sun.jndi.ldap.LdapClientFactory.createPooledConnection(LdapClientFactory.java:71) 
 10:57:42.212 at java.naming/com.sun.jndi.ldap.pool.Connections.createConnection(Connections.java:185) 
 10:57:42.212 at java.naming/com.sun.jndi.ldap.pool.Connections.getAvailableConnection(Connections.java:151) 
 10:57:42.212 at java.naming/com.sun.jndi.ldap.pool.Pool.getOrCreatePooledConnection(Pool.java:189) 
 10:57:42.212 at java.naming/com.sun.jndi.ldap.pool.Pool.getPooledConnection(Pool.java:148) 
 10:57:42.212 at java.naming/com.sun.jndi.ldap.LdapPoolManager.getLdapClient(LdapPoolManager.java:345) 
 10:57:42.213 at java.naming/com.sun.jndi.ldap.LdapClient.getInstance(LdapClient.java:1612) 
 10:57:42.213 at java.naming/com.sun.jndi.ldap.LdapCtx.connect(LdapCtx.java:2848) 
 10:57:42.213 at java.naming/com.sun.jndi.ldap.LdapCtx.<init>(LdapCtx.java:349) 
 10:57:42.213 at java.naming/com.sun.jndi.ldap.LdapCtxFactory.getLdapCtxFromUrl(LdapCtxFactory.java:229) 
 10:57:42.213 at java.naming/com.sun.jndi.ldap.LdapCtxFactory.getUsingURL(LdapCtxFactory.java:189) 
 10:57:42.213 at java.naming/com.sun.jndi.ldap.LdapCtxFactory.getUsingURLs(LdapCtxFactory.java:247) 
 10:57:42.213 at java.naming/com.sun.jndi.ldap.LdapCtxFactory.getLdapCtxInstance(LdapCtxFactory.java:154) 
 10:57:42.213 at java.naming/com.sun.jndi.ldap.LdapCtxFactory.getInitialContext(LdapCtxFactory.java:84) 
 10:57:42.213 at java.naming/javax.naming.spi.NamingManager.getInitialContext(NamingManager.java:520) 
 10:57:42.213 at java.naming/javax.naming.InitialContext.getDefaultInitCtx(InitialContext.java:305) 
 10:57:42.213 at java.naming/javax.naming.InitialContext.init(InitialContext.java:236) 
 10:57:42.213 at java.naming/javax.naming.ldap.InitialLdapContext.<init>(InitialLdapContext.java:154) 
 10:57:42.213 at org.keycloak.storage.ldap.idm.store.ldap.LDAPContextManager.createLdapContext(LDAPContextManager.java:80) 
 10:57:42.213 at org.keycloak.storage.ldap.idm.store.ldap.LDAPContextManager.getLdapContext(LDAPContextManager.java:100) 
 10:57:42.213 at org.keycloak.storage.ldap.idm.store.ldap.LDAPOperationManager.execute(LDAPOperationManager.java:701) 
 10:57:42.213 at org.keycloak.storage.ldap.idm.store.ldap.LDAPOperationManager.execute(LDAPOperationManager.java:696) 
 10:57:42.213 at org.keycloak.storage.ldap.idm.store.ldap.LDAPOperationManager.lookupById(LDAPOperationManager.java:402) 
 10:57:42.213 ... 71 more 
 10:57:42.213  Caused by: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target 
 10:57:42.213 at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:130) 
 10:57:42.213 at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:378) 
 10:57:42.213 at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:321) 
 10:57:42.213 at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:316) 
 10:57:42.213 at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(CertificateMessage.java:647) 
 10:57:42.213 at java.base/sun.security.ssl.CertificateStatus$CertificateStatusConsumer.consume(CertificateStatus.java:296) 
 10:57:42.213 at java.base/sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:393) 
 10:57:42.213 at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:476) 
 10:57:42.213 at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:447) 
 10:57:42.213 at java.base/sun.security.ssl.TransportContext.dispatch(TransportContext.java:201) 
 10:57:42.213 at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:172) 
 10:57:42.213 at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1506) 
 10:57:42.213 at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1421) 
 10:57:42.213 at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:455) 
 10:57:42.213 at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:426) 
 10:57:42.213 at java.naming/com.sun.jndi.ldap.Connection.initialSSLHandshake(Connection.java:365) 
 10:57:42.213 at java.naming/com.sun.jndi.ldap.Connection.createSocket(Connection.java:290) 
 10:57:42.213 at java.naming/com.sun.jndi.ldap.Connection.<init>(Connection.java:232) 
 10:57:42.213 ... 95 more 
 10:57:42.213  Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target 
 10:57:42.213 at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:388) 
 10:57:42.213 at java.base/sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:271) 
 10:57:42.213 at java.base/sun.security.validator.Validator.validate(Validator.java:256) 
 10:57:42.213 at java.base/sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:230) 
 10:57:42.213 at java.base/sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:132) 
 10:57:42.213 at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(CertificateMessage.java:631) 
 10:57:42.213 ... 108 more 
 10:57:42.213  Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target 
 10:57:42.213 at java.base/sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:148) 
 10:57:42.213 at java.base/sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:129) 
 10:57:42.213 at java.base/java.security.cert.CertPathBuilder.build(CertPathBuilder.java:297) 
 10:57:42.213 at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:383) 
 10:57:42.213 ... 113 more 
```

### How to Reproduce?

Connect to a ldap Server with Port 636 and ldaps:// and make an export with cli

d:\keycloak>.\keycloak-24.0.4\bin\kc.bat export --verbose --users same_file --file export/realmName-2024-05-08-10-57-32.json --realm realmName  

### Anything else?

my test server support ldap:// and ldaps:// on the settings dialog i can test both successful

Without ldapS://

![image](https://github.com/keycloak/keycloak/assets/21671762/354be80d-b8b7-4f8c-9fbe-51ea1ff3d814)

With ldapS://

![image](https://github.com/keycloak/keycloak/assets/21671762/6c49fd0a-28d6-4f6e-b45c-09aa8e1a5faf)

But an export works only with Port ldap/389 and not with ldaps/636

I've sent https://github.com/keycloak/keycloak/pull/29400 to allow using vault options when running the `export` command.

@cyberrranger I run a local test using a setup that I think is similar to yours:

* Server started using `--truststore-paths=/pedroigor/development/conf/keycloak/realms/dev/cert.pem --vault=file --vault-dir=/tmp/myvault`
* LDAP provider configured with credentials using `${vault.ldapcredentials}` and using `ldaps`

Trying to run `export` without the truststore will fail and the error I'm getting is:

```
Caused by: java.net.SocketException: Connection or outbound has closed
        at java.base/sun.security.ssl.SSLSocketImpl$AppOutputStream.write(SSLSocketImpl.java:1296)
        at java.base/java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:81)
        at java.base/java.io.BufferedOutputStream.flush(BufferedOutputStream.java:142)
        at java.naming/com.sun.jndi.ldap.Connection.writeRequest(Connection.java:415)
        at java.naming/com.sun.jndi.ldap.Connection.writeRequest(Connection.java:388)
        at java.naming/com.sun.jndi.ldap.LdapClient.ldapBind(LdapClient.java:359)
        at java.naming/com.sun.jndi.ldap.LdapClient.authenticate(LdapClient.java:214)
```

I'm not sure why I'm not getting the usual cert path validation exception ...

Running without the `vault-dir` option will cause an error because the server won't be able to authenticate due to invalid credentials as the credential wasn't resolved from the vault.

This https://github.com/keycloak/keycloak/pull/29400 should fix the latter.

--- a/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/AbstractExportImportCommand.java
+++ b/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/AbstractExportImportCommand.java
@@ -55,7 +55,6 @@ public abstract class AbstractExportImportCommand extends AbstractStartCommand i
                         optionCategory != OptionCategory.HOSTNAME_V1 &&
                         optionCategory != OptionCategory.HOSTNAME_V2 &&
                         optionCategory != OptionCategory.METRICS &&
-                        optionCategory != OptionCategory.VAULT &&
                         optionCategory != OptionCategory.SECURITY &&
                         optionCategory != OptionCategory.CACHE &&
                         optionCategory != OptionCategory.HEALTH).collect(Collectors.toList());

